route/v1/types.go

```diff
@@ -139,8 +139,90 @@ type RouteSpec struct {
 	// +kubebuilder:validation:Enum=None;Subdomain;""
 	// +kubebuilder:default=None
 	WildcardPolicy WildcardPolicyType `json:"wildcardPolicy,omitempty" protobuf:"bytes,7,opt,name=wildcardPolicy"`
+
+	// httpHeaders defines policy for HTTP headers.
+	//
+	// If this field is empty, the default values are used.
+	//
+	// +optional
+	HTTPHeaders *RouteHTTPHeaders `json:"httpHeaders,omitempty" protobuf:"bytes,9,opt,name=httpHeaders"`
+}
+
+// RouteHTTPHeaders defines policy for HTTP headers.
+type RouteHTTPHeaders struct {
+	// actions specifies options for performing action on headers.
+	// For Setting Headers one can provide a value for the header whereas for deleting one needs to provide header name.
+	// +optional
+	Actions RouteHTTPHeaderActions `json:"actions,omitempty" protobuf:"bytes,1,opt,name=actions"`
+}
+
+// RouteHTTPHeaderActions defines whether we want to set Response or Request Headers.
+type RouteHTTPHeaderActions struct {
+	// Response is a list of http response headers which either need to set or deleted.
+	Response []RouteHTTPHeader `json:"response,omitempty" protobuf:"bytes,1,opt,name=response"`
+	// Request is a list of http request headers which either need to set or deleted.
+	Request []RouteHTTPHeader `json:"request,omitempty" protobuf:"bytes,2,opt,name=request"`
```

Comment:
```suggestion
	// response is a list of HTTP response headers to set or delete.
	Response []RouteHTTPHeader `json:"response,omitempty" protobuf:"bytes,1,opt,name=response"`
	// request is a list of HTTP request headers to set or delete.
	Request []RouteHTTPHeader `json:"request,omitempty" protobuf:"bytes,2,opt,name=request"`
```
Document the following:

* The significance of ordering for values in these fields.
* Interactions with the corresponding configuration on the IngressController.
* Interaction with session cookies.
* Interaction with the `haproxy.router.openshift.io/hsts_header` route annotation.

Reply:
@miheer, you addressed the significance of ordering with godoc on the `response` and `request` fields, and you addressed interactions with IngressController config and HSTS with godoc on the `actions` field, but I don't see where you have addressed interactions with session cookies.

Reply:
My previous comment still hasn't been addressed.

Reply:
https://github.com/openshift/api/compare/8e4d36aaebb33d8bc4e10bd2e5cfc2efeb803ff3..3ed233511a8dd9484a47a184709f6a75d32ce70e added documentation about session cookies.  Thanks!